
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>ggd get-files &#8212; GGD  documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/style.css" />
    <link rel="stylesheet" type="text/css" href="_static/font-awesome-4.7.0/css/font-awesome.min.css" />
    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/language_data.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="ggd pkg-info" href="pkg-info.html" />
    <link rel="prev" title="ggd list" href="list.html" />
    
    <link href="https://fonts.googleapis.com/css?family=Lato|Raleway" rel="stylesheet">
    <link href="https://fonts.googleapis.com/css?family=Inconsolata" rel="stylesheet">

    <meta name="msapplication-TileColor" content="#ffffff">
    <meta name="msapplication-TileImage" content="_static/ms-icon-144x144.png">

    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/selectize.js/0.12.6/css/selectize.bootstrap3.min.css">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.3.1/css/bootstrap.min.css">
    <script src="https://cdnjs.cloudflare.com/ajax/libs/datatables/1.10.21/js/jquery.dataTables.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/selectize.js/0.12.6/js/standalone/selectize.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.3.1/js/bootstrap.bundle.min.js"></script>




  </head><body>
  <div class="document">
    
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<p class="logo">
  <a href="index.html">
    <img class="logo" src="_static/logo/GoGetData_name_logo.png" alt="Logo"/>
    
  </a>
</p>








<h3>Navigation</h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="quick-start.html">GGD Quick Start</a></li>
<li class="toctree-l1"><a class="reference internal" href="using-ggd.html">Using GGD</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="GGD-CLI.html">GGD Commands</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="ggd-search.html">ggd search</a></li>
<li class="toctree-l2"><a class="reference internal" href="install.html">ggd install</a></li>
<li class="toctree-l2"><a class="reference internal" href="predict-path.html">ggd predict-path</a></li>
<li class="toctree-l2"><a class="reference internal" href="uninstall.html">ggd uninstall</a></li>
<li class="toctree-l2"><a class="reference internal" href="list.html">ggd list</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">ggd get-files</a></li>
<li class="toctree-l2"><a class="reference internal" href="pkg-info.html">ggd pkg-info</a></li>
<li class="toctree-l2"><a class="reference internal" href="show-env.html">ggd show-env</a></li>
<li class="toctree-l2"><a class="reference internal" href="make-recipe.html">ggd make-recipe</a></li>
<li class="toctree-l2"><a class="reference internal" href="check-recipe.html">ggd check-recipe</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="contribute.html">Contribute</a></li>
<li class="toctree-l1"><a class="reference internal" href="private_recipes.html">Private Recipes</a></li>
<li class="toctree-l1"><a class="reference internal" href="recipes.html">Available Data Packages</a></li>
</ul>

<ul>
  
  <li class="toctree-l1"><a href="https://github.com/gogetdata/ggd-recipes">ggd-recipes @ Github</a></li>
  
  <li class="toctree-l1"><a href="https://github.com/gogetdata/ggd-cli">ggd-cli @ Github</a></li>
  
</ul>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="ggd-get-files">
<span id="id1"></span><h1>ggd get-files<a class="headerlink" href="#ggd-get-files" title="Permalink to this headline">¶</a></h1>
<p>[<a class="reference internal" href="index.html#home-page"><span class="std std-ref">Click here to return to the home page</span></a>]</p>
<p>The <code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">get-files</span></code> command is used to get/list the files associated with a ggd data package that has been installed on your system
using <code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">install</span></code>.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If the data package has not been installed on your system <code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">get-files</span></code> will not be able to list the files.</p>
</div>
<p>This tool is used to find the location of a file(s) downloaded using ggd. For example, if you downloaded the GRCh37
reference genome using <code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">install</span> <span class="pre">grch37-reference-genome-ensembl-v1</span></code> and you want the fasta file and the fai file you would
run <code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">get-files</span> <span class="pre">grch37-reference-genome-ensembl-v1</span></code>. The file locations would then be provided via STDOUT.</p>
<div class="section" id="using-ggd-get-files">
<h2>Using ggd get-files<a class="headerlink" href="#using-ggd-get-files" title="Permalink to this headline">¶</a></h2>
<p>Use <code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">get-files</span></code> to list the files associated with a previously installed ggd package.
Running <code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">get-files</span> <span class="pre">-h</span></code> will give you the following message:</p>
<p>get-files arguments:</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 35%" />
<col style="width: 65%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>get-files</p></th>
<th class="head"><p>Get a list of file(s) for a specific installed ggd package</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-h</span></code>, <code class="docutils literal notranslate"><span class="pre">--help</span></code></p></td>
<td><p>show this help message and exit</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">name</span></code></p></td>
<td><p><strong>Required</strong>  recipe name</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>, <code class="docutils literal notranslate"><span class="pre">--channel</span></code></p></td>
<td><p>(Optional) The ggd channel of the recipe to find. (Default = genomics)</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-s</span></code>, <code class="docutils literal notranslate"><span class="pre">--species</span></code></p></td>
<td><p>(Optional) species recipe is for. Use ‘*’ for any species</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-g</span></code> , <code class="docutils literal notranslate"><span class="pre">--genome-build</span></code></p></td>
<td><p>(Optional) genome build the recipe is for. Use ‘*’ for any genome build.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-v</span></code>, <code class="docutils literal notranslate"><span class="pre">--version</span></code></p></td>
<td><p>(Optional) pattern to match the version of the file desired.
Use ‘*’ for any version</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-p</span></code>, <code class="docutils literal notranslate"><span class="pre">--pattern</span></code></p></td>
<td><p>(Optional) pattern to match the name of the file desired. To
list all files for a ggd package, do not use the -p option.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--prefix</span></code></p></td>
<td><p>(Optional) The name or the full directory path to an
conda environment where a ggd recipe is stored. (Only
needed if not getting file paths for files in the
current conda environment)</p></td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">get-files</span></code> can be used in a script that uses a data package. This usually means you use the <code class="code docutils literal notranslate"><span class="pre">-p</span></code> flag to
get a specific pattern. See <span class="xref std std-ref">example 3 below</span> for an example of how to use the output.</p>
</div>
<div class="section" id="additional-argument-explanation">
<h3>Additional argument explanation:<a class="headerlink" href="#additional-argument-explanation" title="Permalink to this headline">¶</a></h3>
<p>Required arguments:</p>
<ul class="simple">
<li><p><em>name:</em> The <code class="code docutils literal notranslate"><span class="pre">name</span></code> field is required and represents the ggd data package name for which to list files.</p></li>
</ul>
<p>Optional arguments for filtering:</p>
<ul class="simple">
<li><p><em>-c:</em> The <code class="code docutils literal notranslate"><span class="pre">-c</span></code> flag is used to set the ggd channel (default: genomics).</p></li>
<li><p><em>-s:</em> The <code class="code docutils literal notranslate"><span class="pre">-s</span></code> flag represents the species of the ggd data package and filters files based on species.</p></li>
<li><p><em>-g:</em> The <code class="code docutils literal notranslate"><span class="pre">-g</span></code> flag represents the genome-build of the ggd data package and filters the list to the specified build.</p></li>
<li><p><em>-v:</em> The <code class="code docutils literal notranslate"><span class="pre">-v</span></code> flag represents the ggd data package version. This flag helps to refine the package to a specific version.</p></li>
<li><p><em>-p:</em> The <code class="code docutils literal notranslate"><span class="pre">-p</span></code> flag is used to list files that have a specific pattern, such as ‘*.fai’ or ‘*.bam’.</p></li>
<li><p><em>–prefix:</em> The <code class="code docutils literal notranslate"><span class="pre">--prefix</span></code> flag is used to designate which conda environment/prefix to get the file from.
<strong>This allows one to store ggd data packages in one environment and access it from another.</strong></p></li>
</ul>
<p>The <strong>name</strong> field is the only required parameter for <code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">get-files</span></code>, however, the other flags are provided to help reduce the
search space and refine the final file list.</p>
</div>
</div>
<div class="section" id="examples">
<h2>Examples<a class="headerlink" href="#examples" title="Permalink to this headline">¶</a></h2>
<div class="section" id="simple-example-with-only-the-name-field">
<h3>1. Simple example with only the name field<a class="headerlink" href="#simple-example-with-only-the-name-field" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ ggd get-files grch37-reference-genome-1000g-v1

  &lt;conda root&gt;/share/ggd/Homo_sapiens/GRCh37/grch37-reference-genome-1000g-v1/1/hs37d5.fa.fai
  &lt;conda root&gt;/share/ggd/Homo_sapiens/GRCh37/grch37-reference-genome-1000g-v1/1/hs37d5.fa
</pre></div>
</div>
</div>
<div class="section" id="example-using-the-s-g-and-p-flags-to-find-the-gzipped-vcf-files-for-a-given-ggd-data-package">
<h3>2. Example using the <code class="docutils literal notranslate"><span class="pre">-s</span></code>, <code class="docutils literal notranslate"><span class="pre">-g</span></code>, and <code class="docutils literal notranslate"><span class="pre">-p</span></code> flags to find the gzipped vcf files for a given ggd data package<a class="headerlink" href="#example-using-the-s-g-and-p-flags-to-find-the-gzipped-vcf-files-for-a-given-ggd-data-package" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ ggd get-files grch37-esp-variants-uw-v1 -s Homo_sapiens -g GRCh37 -p *.vcf.gz

 &lt;conda root&gt;/share/ggd/Homo_sapiens/GRCh37/grch37-esp-variants-uw-v1/1/grch37-esp-variants-uw-v1.vcf.gz


$ ggd get-files grch37-esp-variants-uw-v1 -s Homo_sapiens -g GRCh37 -p *.vcf.gz.tbi

 &lt;conda root&gt;/share/ggd/Homo_sapiens/GRCh37/grch37-esp-variants-uw-v1/1/ESP6500SI.all.snps_indels.tidy.vcf.gz.tbi
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><code class="docutils literal notranslate"><span class="pre">&lt;conda</span> <span class="pre">root&gt;</span></code> represents the root directory for the local conda repository on your system.</p>
</div>
</div>
<div class="section" id="example-of-using-the-prefix-flag-to-get-data-files-from-a-different-conda-environment">
<h3>3. Example of using the <code class="code docutils literal notranslate"><span class="pre">--prefix</span></code> flag to get data files from a different conda environment<a class="headerlink" href="#example-of-using-the-prefix-flag-to-get-data-files-from-a-different-conda-environment" title="Permalink to this headline">¶</a></h3>
<p><code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">get-files</span></code> allows one to access ggd data packages stored in different conda environments using the <code class="code docutils literal notranslate"><span class="pre">--prefix</span></code> flag.
In this example we are going to:</p>
<blockquote>
<div><ol class="upperalpha simple">
<li><p>Get the files for the ggd data package <code class="code docutils literal notranslate"><span class="pre">grch37-eiee-genes-ostrander-v1</span></code> stored in the conda environment <code class="code docutils literal notranslate"><span class="pre">data:</span></code></p></li>
<li><p>From the <code class="code docutils literal notranslate"><span class="pre">grch37-eiee-genes-ostrander-v1</span></code> ggd data package in the <code class="code docutils literal notranslate"><span class="pre">data</span></code> conda environment, get
the <code class="code docutils literal notranslate"><span class="pre">grch37-eiee-genes-ostrander-v1.bed.gz</span></code> file only.</p></li>
<li><p>Get the <code class="code docutils literal notranslate"><span class="pre">grch37-eiee-genes-ostrander-v1.bed.gz</span></code> data file as in B, but use a wild card within the <code class="code docutils literal notranslate"><span class="pre">-p</span></code> pattern flag</p></li>
</ol>
</div></blockquote>
<ol class="upperalpha simple">
<li></li>
</ol>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ ggd get-files grch37-eiee-genes-ostrander-v1 --prefix data

  &lt;conda root&gt;/share/ggd/Homo_sapiens/GRCh37/grch37-eiee-genes-ostrander-v1/1/grch37-eiee-genes-ostrander-v1.bed.gz.tbi
  &lt;conda root&gt;/share/ggd/Homo_sapiens/GRCh37/grch37-eiee-genes-ostrander-v1/1/grch37-eiee-genes-ostrander-v1.complement.bed.gz.tbi
  &lt;conda root&gt;/share/ggd/Homo_sapiens/GRCh37/grch37-eiee-genes-ostrander-v1/1/grch37-eiee-genes-ostrander-v1.bed.gz
  &lt;conda root&gt;/share/ggd/Homo_sapiens/GRCh37/grch37-eiee-genes-ostrander-v1/1/grch37-eiee-genes-ostrander-v1.complement.bed.gz
</pre></div>
</div>
<ol class="upperalpha simple" start="2">
<li></li>
</ol>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ ggd get-files grch37-eiee-genes-ostrander-v1 --prefix data -p grch37-eiee-genes-ostrander-v1.bed.gz

  &lt;conda root&gt;/share/ggd/Homo_sapiens/GRCh37/grch37-eiee-genes-ostrander-v1/1/grch37-eiee-genes-ostrander-v1.bed.gz
</pre></div>
</div>
<ol class="upperalpha simple" start="3">
<li></li>
</ol>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ ggd get-files grch37-eiee-genes-ostrander-v1 --prefix data -p *v1.bed.gz

  &lt;conda root&gt;/share/ggd/Homo_sapiens/GRCh37/grch37-eiee-genes-ostrander-v1/1/grch37-eiee-genes-ostrander-v1.bed.gz
</pre></div>
</div>
</div>
<div class="section" id="use-the-output-of-the-get-files-command-as-input-to-a-shell-command">
<span id="ggd-get-files-example-4"></span><h3>4. Use the output of the get-files command as input to a shell command<a class="headerlink" href="#use-the-output-of-the-get-files-command-as-input-to-a-shell-command" title="Permalink to this headline">¶</a></h3>
<p>Installing data is one necessary part of genomic data management, however, using the data is the vital part. Therefore,
the <code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">get-files</span></code> command was developed to provide easy access and use to the installed ggd data packages.
Simply stated, <code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">get-files</span></code> can be used to access a desired file from an installed ggd data package in a command,
in a script, as a parameter, etc.</p>
<p><strong>In order to use a data file you will need to use the</strong> <code class="code docutils literal notranslate"><span class="pre">-p</span></code> <strong>flag in order to get a single file as an output.</strong></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you are in the conda package where the data is stored you can use the environment variables created for each data
package as another way to access the files. (To see the available environment variables for the active conda environment
run <code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">show-env</span></code> or <code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">list</span></code>) If the data you are accessing is stored in a different conda environment then
the data environment variables will not be available for use. You will need to use <code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">get-files</span></code> with the
<code class="code docutils literal notranslate"><span class="pre">--prefix</span></code> flag set (See example 3 above).</p>
</div>
<ol class="upperalpha simple">
<li><p>Assigning the path to an environment variable and using it in the shell.</p></li>
</ol>
<blockquote>
<div><p>Here we will get and use the <code class="code docutils literal notranslate"><span class="pre">.bed.gz</span></code> file from the <code class="code docutils literal notranslate"><span class="pre">hg19-cpg-islands-ucsc-v1</span></code> ggd data packages</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ cpg_islands=&quot;$(ggd get-files hg19-cpg-islands-ucsc-v1 -p *.bed.gz)&quot;
</pre></div>
</div>
<p>To check that the file is in this new variable called <code class="code docutils literal notranslate"><span class="pre">cpg_islands</span></code> run:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ echo $cpg_islands
</pre></div>
</div>
<p>You can now use this cpg_islands variable in a script.</p>
<ol class="arabic simple">
<li><p>Use tabix to get CpG info for a specific genomic region</p></li>
</ol>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ tabix $cpg_islands chr6:150284682-152129771

  chr6  150284682 150286515 CpG: 196
  chr6  150311256 150312369 CpG: 85
  chr6  150326011 150326802 CpG: 67
  chr6  150335525 150336278 CpG: 64
  chr6  150358872 150359394 CpG: 51
  chr6  150378838 150379048 CpG: 16
  chr6  150389943 150390558 CpG: 55
  chr6  150463771 150465002 CpG: 127
  chr6  150920904 150922146 CpG: 129
  chr6  151186747 151188112 CpG: 151
  chr6  151412003 151412339 CpG: 25
  chr6  151560766 151560993 CpG: 16
  chr6  151561283 151562550 CpG: 132
  chr6  151646668 151646958 CpG: 25
  chr6  151662605 151663056 CpG: 42
  chr6  151711094 151712829 CpG: 195
  chr6  151773043 151774070 CpG: 96
  chr6  151814980 151815527 CpG: 64
  chr6  152128822 152129771 CpG: 89
</pre></div>
</div>
</div></blockquote>
<ol class="arabic simple" start="2">
<li><p>You can subset a vcf file by cpg island regions using bedtools</p></li>
</ol>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ bedtools intersect -a &lt;your-vcf-file&gt; -b $cpg_islands -wa &gt; vcf_file_cpg_subset.vcf
</pre></div>
</div>
</div></blockquote>
<ol class="arabic simple" start="3">
<li><p>And many other options you could think of…</p></li>
</ol>
</div></blockquote>
<ol class="upperalpha simple" start="2">
<li><p>Piping the output from <code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">get-files</span></code> to a command</p></li>
</ol>
<blockquote>
<div><p>Usinsg the example above of subsetting a vcf file by the cpg island regions using bedtools</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ggd</span> <span class="n">get</span><span class="o">-</span><span class="n">files</span> <span class="n">hg19</span><span class="o">-</span><span class="n">cpg</span><span class="o">-</span><span class="n">islands</span><span class="o">-</span><span class="n">ucsc</span><span class="o">-</span><span class="n">v1</span> <span class="o">-</span><span class="n">p</span> <span class="o">*.</span><span class="n">bed</span><span class="o">.</span><span class="n">gz</span> \
  <span class="o">|</span> <span class="n">bedtools</span> <span class="n">intersect</span> <span class="o">-</span><span class="n">a</span> <span class="o">&lt;</span><span class="n">your</span><span class="o">-</span><span class="n">vcf</span><span class="o">-</span><span class="n">file</span><span class="o">&gt;</span> <span class="o">-</span><span class="n">b</span> <span class="o">-</span> <span class="o">-</span><span class="n">wa</span> <span class="o">&gt;</span> <span class="n">vcf_file_cpg_subset</span><span class="o">.</span><span class="n">vcf</span>
</pre></div>
</div>
</div></blockquote>
<p>There are many other ways not listed here to use get-files to use the installed ggd data files in a script, shell command,
workflow, etc.</p>
<p>If your data is stored in a different conda environment you could easily add the <code class="code docutils literal notranslate"><span class="pre">--prefix</span></code> flag to
the command.</p>
<p>If you have examples you would like added or you would like to share, let us know and we can add it to the docs.</p>
</div>
</div>
</div>


          </div>
          
        </div>
      </div>
    <div class="clearer"></div>
  </div>
    <div class="footer">
      &copy;2016-2020, The GoGetData team.
      
      |
      <a href="_sources/list-file.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>