
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>ggd make-recipe &#8212; ggd  documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/style.css" />
    <link rel="stylesheet" type="text/css" href="_static/font-awesome-4.7.0/css/font-awesome.min.css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="ggd check-recipe" href="check-recipe.html" />
    <link rel="prev" title="ggd show-env" href="show-env.html" />
    <link href="https://fonts.googleapis.com/css?family=Lato|Raleway" rel="stylesheet"> 
    <link rel="stylesheet" type="text/css" href="_static/font-awesome-4.7.0/css/font-awesome.min.css">
    <link rel="stylesheet" type="text/css" href="_static/style.css">
    <link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/v/dt/jqc-1.12.3/dt-1.10.12/datatables.min.css"/>
    <link href="https://fonts.googleapis.com/css?family=Inconsolata" rel="stylesheet">

    <link rel="apple-touch-icon" sizes="57x57" href="/apple-icon-57x57.png">
    <link rel="apple-touch-icon" sizes="60x60" href="/apple-icon-60x60.png">
    <link rel="apple-touch-icon" sizes="72x72" href="/apple-icon-72x72.png">
    <link rel="apple-touch-icon" sizes="76x76" href="/apple-icon-76x76.png">
    <link rel="apple-touch-icon" sizes="114x114" href="/apple-icon-114x114.png">
    <link rel="apple-touch-icon" sizes="120x120" href="/apple-icon-120x120.png">
    <link rel="apple-touch-icon" sizes="144x144" href="/apple-icon-144x144.png">
    <link rel="apple-touch-icon" sizes="152x152" href="/apple-icon-152x152.png">
    <link rel="apple-touch-icon" sizes="180x180" href="/apple-icon-180x180.png">
    <link rel="icon" type="image/png" sizes="192x192"  href="/android-icon-192x192.png">
    <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
    <link rel="icon" type="image/png" sizes="96x96" href="/favicon-96x96.png">
    <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
    <link rel="manifest" href="/manifest.json">
    <meta name="msapplication-TileColor" content="#ffffff">
    <meta name="msapplication-TileImage" content="/ms-icon-144x144.png">
    <meta name="theme-color" content="#ffffff">
    <script type="text/javascript" src="https://cdn.datatables.net/1.10.19/js/jquery.dataTables.min.js"></script>

    <script type="text/javascript" class="init">
    $(document).ready(function() {
        $('#recipestable').DataTable( {
            "lengthMenu": [[25, 50, -1], [25, 50, "All"]]
        } );
    } );
    </script>

  </head><body>
  <div class="document">
    
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<p class="logo">
  <a href="index.html">
    <img class="logo" src="_static/logo/gogetdata_temp.png" alt="Logo"/>
    
  </a>
</p>








<h3>Navigation</h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="quick-start.html">GGD Quick Start</a></li>
<li class="toctree-l1"><a class="reference internal" href="using-ggd.html">Using GGD</a></li>
<li class="toctree-l1"><a class="reference internal" href="ggd-search.html">ggd search</a></li>
<li class="toctree-l1"><a class="reference internal" href="install.html">ggd install</a></li>
<li class="toctree-l1"><a class="reference internal" href="uninstall.html">ggd uninstall</a></li>
<li class="toctree-l1"><a class="reference internal" href="list-file.html">ggd list-files</a></li>
<li class="toctree-l1"><a class="reference internal" href="pkg-info.html">ggd pkg-info</a></li>
<li class="toctree-l1"><a class="reference internal" href="show-env.html">ggd show-env</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">ggd make-recipe</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#using-ggd-make-recipe">Using ggd make-recipe</a></li>
<li class="toctree-l2"><a class="reference internal" href="#examples">Examples</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="check-recipe.html">ggd check-recipe</a></li>
<li class="toctree-l1"><a class="reference internal" href="contribute.html">Contribute</a></li>
<li class="toctree-l1"><a class="reference internal" href="recipes.html">Available data packages</a></li>
</ul>

<ul>
  
  <li class="toctree-l1"><a href="https://github.com/gogetdata/ggd-recipes">ggd-recipes @ Github</a></li>
  
  <li class="toctree-l1"><a href="https://github.com/gogetdata/ggd-cli">ggd-cli @ Github</a></li>
  
</ul>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="ggd-make-recipe">
<span id="id1"></span><h1>ggd make-recipe<a class="headerlink" href="#ggd-make-recipe" title="Permalink to this headline">¶</a></h1>
<p>[<a class="reference internal" href="index.html#home-page"><span class="std std-ref">Click here to return to the home page</span></a>]</p>
<p>ggd make-recipe is used to create a ggd data recipe from a bash script which contains the information on
extracting and processing the data.</p>
<ul class="simple">
<li><strong>recipe</strong>: A data recipe is directory containing a set of files that contain information about the recipe.
This includes: A meta.yaml file, which is the meta data information for the soon to be ggd data package;
a post-link script, which contains the information about file and data management; and a recipe script, which
contains the information on how to get the data and how to process it.</li>
<li><strong>package</strong>: A data package is created from building the ggd data recipe. It is a bgzipped tar file that contains
the built data recipe and additional metadata information for conda system handling.</li>
</ul>
<p><code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">make-recipe</span></code> takes a bash script created by you and turns it into a data recipe. This data recipe will then be
turned into a data package using <a class="reference internal" href="check-recipe.html#ggd-check-recipe"><span class="std std-ref">ggd check-recipe</span></a>. Finally, the new data package will
be added to the ggd repo and ggd conda channel through a continuous integration system. For more details see
the <a class="reference internal" href="contribute.html#make-data-packages"><span class="std std-ref">contribute</span></a> documentation.</p>
<p>The first step in this process is to create a bash script with instructions on downloading and data processing,
then using <code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">make-recipe</span></code> to create a ggd data recipe.</p>
<div class="section" id="using-ggd-make-recipe">
<h2>Using ggd make-recipe<a class="headerlink" href="#using-ggd-make-recipe" title="Permalink to this headline">¶</a></h2>
<p>Creating a ggd recipe is easy using the <code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">make-recipe</span></code> tool.
Running <code class="code docutils literal notranslate"><span class="pre">ggd</span> <span class="pre">make-recipe</span> <span class="pre">-h</span></code> will give you the following help message:</p>
<p>make-recipe arguments:</p>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">-h</span>, <span class="option">--help</span></kbd></td>
<td>show this help message and exit</td></tr>
<tr><td class="option-group">
<kbd><span class="option">/name</span></kbd></td>
<td>(Positional) Name of recipe
(‘/’ indicates a placeholder and is not part of the argument name)</td></tr>
<tr><td class="option-group">
<kbd><span class="option">/script</span></kbd></td>
<td>(Positional) Bash script that contains the commands that build the recipe
(‘/’ indicates a placeholder and is not part of the argument name)</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-c</span>, <span class="option">--channel</span></kbd></td>
<td>(Optional) The ggd channel to use. (Default = genomics)</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-d <var>DEPENDENCY</var></span>, <span class="option">--dependency <var>DEPENDENCY</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>any software dependencies (in bioconda, conda-forge) or
data-dependency (in ggd). May be as many times as needed.</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-e <var>EXTRA_FILE</var></span>, <span class="option">--extra-file <var>EXTRA_FILE</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>any files that the recipe creates that are not a *.gz
and *.gz.tbi pair. May be used more than once.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-p</span>, <span class="option">--platform</span></kbd></td>
<td>(Optional) Whether to use noarch as the platfrom or the system
platform. If set to ‘none’ the system platform will be
used. (Default = noarch. Noarch means no architecture
and is platform agnostic.)</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-s</span>, <span class="option">--species</span></kbd></td>
<td>(Required) Species recipe is for</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-g <var>GENOME_BUILD</var></span>, <span class="option">--genome-build <var>GENOME_BUILD</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>(Required) Genome-build the recipe is for</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--authors <var>AUTHORS</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>(Required) Authors of the recipe</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-g<var>v</var></span>, <span class="option">--ggd_version</span></kbd></td>
</tr>
<tr><td>&#160;</td><td>(Required) The version of the ggd package. (First time package = 1,
updated package &gt; 1)</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-d<var>v</var></span>, <span class="option">--data_version</span></kbd></td>
</tr>
<tr><td>&#160;</td><td>(Required) The version of the data (itself) being downloaded and
processed (EX: dbsnp-127)</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--summary <var>SUMMARY</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>(Required) A comment describing the recipe</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-k <var>KEYWORD</var></span>, <span class="option">--keyword <var>KEYWORD</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>(Required) A keyword to associate with the recipe. may be specified
more that once.</td></tr>
</tbody>
</table>
<div class="section" id="additional-argument-explanation">
<h3>Additional argument explanation:<a class="headerlink" href="#additional-argument-explanation" title="Permalink to this headline">¶</a></h3>
<p>Required arguments:</p>
<ul class="simple">
<li><em>-s:</em> The <code class="code docutils literal notranslate"><span class="pre">-s</span></code> flag is used to declare the species of the data recipe.</li>
<li><em>-g:</em> The <code class="code docutils literal notranslate"><span class="pre">-g</span></code> flag is used to declare the genome-build of the data recipe.</li>
<li><em>–authors:</em> The <code class="code docutils literal notranslate"><span class="pre">--authors</span></code> flag is used to declare the authors of the recipe.</li>
<li><em>-gv:</em> The <code class="code docutils literal notranslate"><span class="pre">-gv</span></code> flag is used to declare the ggd recipe version. (1 for first time recipe, and 2+ for updated recipes)</li>
<li><em>-dv:</em> The <code class="code docutils literal notranslate"><span class="pre">-dv</span></code> flag is used to declare the version of the data being downloaded and processed. If a version is not
available for the specific data, use something that can identify the data uniquely such as when the data was created.</li>
<li><em>–summary:</em> The <code class="code docutils literal notranslate"><span class="pre">--summary</span></code> flag is used to provide a brief summary/description of the recipe.</li>
<li><em>-k:</em> The <code class="code docutils literal notranslate"><span class="pre">-k</span></code> flag is used to declare keywords associated with the data and recipe. If there are multiple keywords, the <cite>-k</cite> flag
should be used for each keywords. (Example: -k ref -k reference)</li>
<li><em>name:</em> <code class="code docutils literal notranslate"><span class="pre">name</span></code> represents the name of the recipe.</li>
<li><em>script:</em> <code class="code docutils literal notranslate"><span class="pre">script</span></code> represents the bash script containing the information on data extraction and processing.</li>
</ul>
<p>Optional arguments:</p>
<ul class="simple">
<li><em>-c:</em> The <code class="code docutils literal notranslate"><span class="pre">-c</span></code> flag is used to declare which ggd channel to use. (genomics is the default)</li>
<li><em>-d:</em> The <code class="code docutils literal notranslate"><span class="pre">-d</span></code> flag is used to declare software dependencies in conda, bioconda, and conda-forge, and data-dependencies in
ggd for creating the package. If there are not dependencies this flag is not needed.</li>
<li><em>-e:</em> The <code class="code docutils literal notranslate"><span class="pre">-e</span></code> flag is used to declare any extra files created during the data processing. An extra file is any file that is
not a .gz and .gz.tbi pair. If your final processed files are .gz and .gz.tbi pair this flag is not needed.</li>
<li><em>-p:</em> The <code class="code docutils literal notranslate"><span class="pre">-p</span></code> flag is used to set the noarch platform or not. By default “noarch” is set, which means the package will be
built and installed with no architecture designation. This means it should be able to build on linux and macOS. If this is not
true you will need to set <code class="code docutils literal notranslate"><span class="pre">-p</span></code> to “none”. The system you are using, linux or macOS will take then take the place of noarch.</li>
</ul>
</div>
</div>
<div class="section" id="examples">
<h2>Examples<a class="headerlink" href="#examples" title="Permalink to this headline">¶</a></h2>
<div class="section" id="a-simple-example-of-creating-a-ggd-recipe">
<h3>1. A simple example of creating a ggd recipe<a class="headerlink" href="#a-simple-example-of-creating-a-ggd-recipe" title="Permalink to this headline">¶</a></h3>
<p>get_data.sh:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nv">genome</span><span class="o">=</span>https://raw.githubusercontent.com/gogetdata/ggd-recipes/master/genomes/Homo_sapiens/hg19/hg19.genome
wget --quiet -O - http://hgdownload.cse.ucsc.edu/goldenpath/hg19/database/gap.txt.gz <span class="se">\</span>
    <span class="p">|</span> gzip -dc <span class="se">\</span>
    <span class="p">|</span> awk -v <span class="nv">OFS</span><span class="o">=</span><span class="s2">&quot;\t&quot;</span> <span class="s1">&#39;BEGIN {print &quot;#chrom\tstart\tend\tsize\ttype\tstrand&quot;} {print $2,$3,$4,$7,$8,&quot;+&quot;}&#39;</span> <span class="se">\</span>
    <span class="p">|</span> gsort /dev/stdin <span class="nv">$genome</span> <span class="se">\</span>
    <span class="p">|</span> bgzip -c &gt; gaps.bed.gz

tabix gaps.bed.gz
</pre></div>
</div>
<p>ggd make-recipe</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ ggd make-recipe -s Homo_sapiens -g hg19 --author mjc --ggd_version <span class="m">1</span> --data_version <span class="m">27</span>-Apr-2009 --summary <span class="s1">&#39;Assembly gaps from USCS&#39;</span> -k gaps -k region gaps get_data.sh
</pre></div>
</div>
<p>This code will create a new ggd recipe:</p>
<blockquote>
<div><ul class="simple">
<li>Directory Name: <strong>hg19-gaps</strong></li>
<li>Files: <strong>meta.yaml</strong>, <strong>post-link.sh</strong>, and <strong>recipe.sh</strong></li>
</ul>
</div></blockquote>
</div>
<div class="section" id="a-more-complex-ggd-recipe">
<h3>2. A more complex ggd recipe<a class="headerlink" href="#a-more-complex-ggd-recipe" title="Permalink to this headline">¶</a></h3>
<p>get_data.sh</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>wget --quiet http://evs.gs.washington.edu/evs_bulk_data/ESP6500SI-V2-SSA137.GRCh38-liftover.snps_indels.vcf.tar.gz

<span class="c1"># extract individual chromosome files</span>
tar -zxf ESP6500SI-V2-SSA137.GRCh38-liftover.snps_indels.vcf.tar.gz

<span class="c1"># combine chromosome files into one</span>
<span class="o">(</span>grep ^# ESP6500SI-V2-SSA137.GRCh38-liftover.chr1.snps_indels.vcf<span class="p">;</span> cat ESP6500SI-V2-SSA137.GRCh38-liftover.chr*.snps_indels.vcf <span class="p">|</span> grep

<span class="c1"># sort the chromosome data according to the .genome file from github</span>
gsort temp.vcf https://raw.githubusercontent.com/gogetdata/ggd-recipes/master/genomes/Homo_sapiens/GRCh37/GRCh37.genome <span class="se">\</span>
    <span class="p">|</span> bgzip -c &gt; ESP6500SI.all.snps_indels.vcf.gz

<span class="c1"># tabix it</span>
tabix -p vcf ESP6500SI.all.snps_indels.vcf.gz

<span class="c1"># get handle for reference file</span>
<span class="nv">reference_fasta</span><span class="o">=</span><span class="s2">&quot;</span><span class="k">$(</span>ggd list-files <span class="s1">&#39;grch37-reference-genome&#39;</span> -s <span class="s1">&#39;Homo_sapiens&#39;</span> -g <span class="s1">&#39;GRCh37&#39;</span> -p <span class="s1">&#39;hs37d5.fa&#39;</span><span class="k">)</span><span class="s2">&quot;</span>

<span class="c1"># get the santizer script</span>
wget --quiet https://raw.githubusercontent.com/arq5x/gemini/00cd627497bc9ede6851eae2640bdaff9f4edfa3/gemini/annotation_provenance/sanit

<span class="c1"># sanitize</span>
zless ESP6500SI.all.snps_indels.vcf.gz <span class="p">|</span> python sanitize-esp.py <span class="p">|</span> bgzip -c &gt; temp.gz
tabix temp.gz

<span class="c1"># decompose with vt</span>
vt decompose -s temp.gz <span class="p">|</span> vt normalize -r <span class="nv">$reference_fasta</span> - <span class="se">\</span>
    <span class="p">|</span> perl -pe <span class="s1">&#39;s/\([EA_|T|AA_]\)AC,Number=R,Type=Integer/\1AC,Number=R,Type=String/&#39;</span> <span class="se">\</span>
    <span class="p">|</span> bgzip -c &gt; ESP6500SI.all.snps_indels.tidy.vcf.gz

tabix ESP6500SI.all.snps_indels.tidy.vcf.gz

<span class="c1"># clean up environment</span>
rm ESP6500SI-V2-SSA137.GRCh38-liftover.snps_indels.vcf.tar.gz
rm ESP6500SI-V2-SSA137.GRCh38-liftover.chr*.snps_indels.vcf

rm ESP6500SI.all.snps_indels.vcf.gz.tbi
rm ESP6500SI.all.snps_indels.vcf.gz

rm temp.gz
rm temp.gz.tbi
rm temp.vcf

rm sanitize-esp.py
</pre></div>
</div>
<p>ggd make-recipe</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ ggd make-recipe -s Homo_sapiens -g GRCh37 --author mjc --ggd_version <span class="m">1</span> --data_version ESP6500SI-V2 --summary <span class="s1">&#39;ESP variants (More Info: http://evs.gs.washington.edu/EVS/#tabs-7)&#39;</span> -k ESP esp-variants get_data.sh
</pre></div>
</div>
<p>This code will create a new ggd recipe:</p>
<blockquote>
<div><ul class="simple">
<li>Directory Name: <strong>grch37-esp-variants</strong></li>
<li>Files: <strong>meta.yaml</strong>, <strong>post-link.sh</strong>, <strong>recipe.sh</strong></li>
</ul>
</div></blockquote>
</div>
</div>
</div>


          </div>
          
        </div>
      </div>
    <div class="clearer"></div>
  </div>
    <div class="footer">
      &copy;2016-2019, The gogetdata team.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.8.4</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/make-recipe.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>