
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>hg19-clinically-associated-variants-ensembl-v1 &#8212; GGD  documentation</title>
    <link rel="stylesheet" href="../../../../../_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="../../../../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../../../../_static/style.css" />
    <link rel="stylesheet" type="text/css" href="../../../../../_static/font-awesome-4.7.0/css/font-awesome.min.css" />
    <script id="documentation_options" data-url_root="../../../../../" src="../../../../../_static/documentation_options.js"></script>
    <script src="../../../../../_static/jquery.js"></script>
    <script src="../../../../../_static/underscore.js"></script>
    <script src="../../../../../_static/doctools.js"></script>
    <script src="../../../../../_static/language_data.js"></script>
    <link rel="index" title="Index" href="../../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../../search.html" />
    
    <link href="https://fonts.googleapis.com/css?family=Lato|Raleway" rel="stylesheet">
    <link href="https://fonts.googleapis.com/css?family=Inconsolata" rel="stylesheet">

    <meta name="msapplication-TileColor" content="#ffffff">
    <meta name="msapplication-TileImage" content="../../../../../_static/ms-icon-144x144.png">

    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/selectize.js/0.12.6/css/selectize.bootstrap3.min.css">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.3.1/css/bootstrap.min.css">
    <script src="https://cdnjs.cloudflare.com/ajax/libs/datatables/1.10.21/js/jquery.dataTables.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/selectize.js/0.12.6/js/standalone/selectize.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.3.1/js/bootstrap.bundle.min.js"></script>




  </head><body>
  <div class="document">
    
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<p class="logo">
  <a href="../../../../../index.html">
    <img class="logo" src="../../../../../_static/logo/GoGetData_name_logo.png" alt="Logo"/>
    
  </a>
</p>








<h3>Navigation</h3>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../../quick-start.html">GGD Quick Start</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../using-ggd.html">Using GGD</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../GGD-CLI.html">GGD Commands</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../contribute.html">Contribute</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../private_recipes.html">Private Recipes</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../recipes.html">Available Data Packages</a></li>
</ul>

<ul>
  
  <li class="toctree-l1"><a href="https://github.com/gogetdata/ggd-recipes">ggd-recipes @ Github</a></li>
  
  <li class="toctree-l1"><a href="https://github.com/gogetdata/ggd-cli">ggd-cli @ Github</a></li>
  
</ul>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../../../../../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="hg19-clinically-associated-variants-ensembl-v1">
<span id="id1"></span><h1>hg19-clinically-associated-variants-ensembl-v1<a class="headerlink" href="#hg19-clinically-associated-variants-ensembl-v1" title="Permalink to this headline">¶</a></h1>
<p><a class="reference external" href="https://anaconda.org/ggd-genomics/hg19-clinically-associated-variants-ensembl-v1"><img alt="downloads" src="https://anaconda.org/ggd-genomics/hg19-clinically-associated-variants-ensembl-v1/badges/downloads.svg" /></a></p>
<p><strong>Liftover</strong> All known clinically associated variants at the time of Ensembl release 99. Any variant in ClinVar classified as probably/likely-pathogenic, pathogenic, drug-response, or histocompatibility are included. Decomposed and Normalized. Liftover from GRCh38 to GRCh37. Remapped from Ensembl GRCh37 to UCSC hg19. Ensembl variant info page can be found at: <a class="reference external" href="https://uswest.ensembl.org/info/genome/variation/index.html">https://uswest.ensembl.org/info/genome/variation/index.html</a></p>
<table class="docutils align-default">
<colgroup>
<col style="width: 12%" />
<col style="width: 88%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p>GGD Package</p></td>
<td><p>hg19-clinically-associated-variants-ensembl-v1</p></td>
</tr>
<tr class="row-even"><td><p>Species</p></td>
<td><p>Homo_sapiens</p></td>
</tr>
<tr class="row-odd"><td><p>Genome Build</p></td>
<td><p>hg19</p></td>
</tr>
<tr class="row-even"><td><p>GGD Channel</p></td>
<td><p>ggd-genomics</p></td>
</tr>
<tr class="row-odd"><td><p>Package Version</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><p>Recipe Author</p></td>
<td><p>mjc</p></td>
</tr>
<tr class="row-odd"><td><p>Data Provider</p></td>
<td><p>Ensembl</p></td>
</tr>
<tr class="row-even"><td><p>Data Version</p></td>
<td><p>release-99_11-7-19</p></td>
</tr>
<tr class="row-odd"><td><p>Genomic File Type</p></td>
<td><p>vcf</p></td>
</tr>
<tr class="row-even"><td><p>Data file coordinate basing</p></td>
<td><p>1-based-inclusive</p></td>
</tr>
<tr class="row-odd"><td><p>Package’s Data Files</p></td>
<td><p>hg19-clinically-associated-variants-ensembl-v1.vcf.gz, hg19-clinically-associated-variants-ensembl-v1.vcf.gz.tbi</p></td>
</tr>
<tr class="row-even"><td><p>Approximate Size of Each Data File</p></td>
<td><p>hg19-clinically-associated-variants-ensembl-v1.vcf.gz: <strong>2.08M</strong>, hg19-clinically-associated-variants-ensembl-v1.vcf.gz.tbi: <strong>105.57K</strong></p></td>
</tr>
<tr class="row-odd"><td><p>Package Keywords</p></td>
<td><p>genetic-varaition, clincal-variants, ClinVar-Associated, Clinically-Associated-Variants, probable-pathogenic-variant, likely-pathogenic-variant, pathogenic-variant, drug-response-variant, histocompatibility, VCF, varaint-info, Ensembl-variantion</p></td>
</tr>
<tr class="row-even"><td><p>Package Dependencies:</p></td>
<td><p>bcftools, crossmap, grch37-reference-genome-ensembl-v1, grch38-reference-genome-ensembl-v1, gsort, hg19-chrom-mapping-ensembl2ucsc-ncbi-v1, hg19-reference-genome-ucsc-v1, htslib, vt, zlib</p></td>
</tr>
<tr class="row-odd"><td><p>Recipe</p></td>
<td><p><a class="reference external" href="https://github.com/gogetdata/ggd-recipes/tree/master/recipes/genomics/Homo_sapiens/hg19/hg19-clinically-associated-variants-ensembl-v1">https://github.com/gogetdata/ggd-recipes/tree/master/recipes/genomics/Homo_sapiens/hg19/hg19-clinically-associated-variants-ensembl-v1</a></p></td>
</tr>
</tbody>
</table>
<div class="section" id="install-with-prefix">
<h2>Install with <code class="docutils literal notranslate"><span class="pre">--prefix</span></code><a class="headerlink" href="#install-with-prefix" title="Permalink to this headline">¶</a></h2>
<p><strong>Prefix install enabled:</strong> <em>True</em></p>
</div>
<div class="section" id="installation">
<h2>Installation<a class="headerlink" href="#installation" title="Permalink to this headline">¶</a></h2>
<p>With ggd installed and an activated ggd channel (see <a class="reference internal" href="../../../../../using-ggd.html#using-ggd"><span class="std std-ref">Using GGD</span></a>), install with:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ggd</span> <span class="n">install</span> <span class="o">-</span><span class="n">c</span> <span class="n">genomics</span> <span class="n">hg19</span><span class="o">-</span><span class="n">clinically</span><span class="o">-</span><span class="n">associated</span><span class="o">-</span><span class="n">variants</span><span class="o">-</span><span class="n">ensembl</span><span class="o">-</span><span class="n">v1</span>
</pre></div>
</div>
</div>
</div>


          </div>
          
        </div>
      </div>
    <div class="clearer"></div>
  </div>
    <div class="footer">
      &copy;2016-2020, The GoGetData team.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 3.2.1</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="../../../../../_sources/recipes/genomics/Homo_sapiens/hg19/hg19-clinically-associated-variants-ensembl-v1/README.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>